File: recipes-service/src/main/java/com/ironhack/recipesservice/model/Recipe.java
package com.ironhack.recipesservice.model;

import com.ironhack.recipesservice.classes.Ingredient;
import com.ironhack.recipesservice.enums.Flavour;
import com.ironhack.recipesservice.enums.FoodType;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "recipes")
public class Recipe {
    @Id
    private Long id;
    private String name;
    private Flavour flavour;
    private FoodType foodType;
    private Integer prepTime;
    private List<Ingredient> ingredients;
    private String preparation;
    private String photo;

    public Recipe(Long id, String name, Flavour flavour, FoodType foodType, Integer prepTime, List<Ingredient> ingredients, String preparation, String photo) {
        this.id = id;
        this.name = name;
        this.flavour = flavour;
        this.foodType = foodType;
        this.prepTime = prepTime;
        this.ingredients = ingredients;
        this.preparation = preparation;
        this.photo = photo;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Flavour getFlavour() {
        return flavour;
    }

    public void setFlavour(Flavour flavour) {
        this.flavour = flavour;
    }

    public FoodType getFoodType() {
        return foodType;
    }

    public void setFoodType(FoodType foodType) {
        this.foodType = foodType;
    }

    public Integer getPrepTime() {
        return prepTime;
    }

    public void setPrepTime(Integer prepTime) {
        this.prepTime = prepTime;
    }

    public List<Ingredient> getIngredients() {
        return ingredients;
    }

    public void setIngredients(List<Ingredient> ingredients) {
        this.ingredients = ingredients;
    }

    public String getPreparation() {
        return preparation;
    }

    public void setPreparation(String preparation) {
        this.preparation = preparation;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }
}

File: recipes-app/recipes-angularapp/src/app/models/recipe/recipe.ts
import { Ingredient } from "../../classes/ingredient"

export class Recipe {

    constructor (
        private _id: number,
        private _flavour: string,
        private _foodType: string,
        private _prepTime: number,
        private _ingredients: Ingredient[] = [],
        private _name: string,
        private _preparation: string,
        private _photo: string
        ) {}

    public get photo(): string {
        return this._photo
    }
    public set photo(value: string) {
        this._photo = value
    }
    public get preparation(): string {
        return this._preparation
    }
    public set preparation(value: string) {
        this._preparation = value
    }
    public get name(): string {
        return this._name
    }
    public set name(value: string) {
        this._name = value
    }
    public get ingredients(): Ingredient[] {
        return this._ingredients
    }
    public set ingredients(value: Ingredient[]) {
        this._ingredients = value
    }
    public get prepTime(): number {
        return this._prepTime
    }
    public set prepTime(value: number) {
        this._prepTime = value
    }
    public get foodType(): string {
        return this._foodType
    }
    public set foodType(value: string) {
        this._foodType = value
    }
    public get flavour(): string {
        return this._flavour
    }
    public set flavour(value: string) {
        this._flavour = value
    }
    public get id(): number {
        return this._id
    }
    public set id(value: number) {
        this._id = value
    }
   
}
    